generators/entity/index.js

```diff
@@ -110,18 +115,44 @@ module.exports = EntityGenerator.extend({
                 this.env.error(chalk.red('The entity name cannot end with \'Detail\''));
             } else if (RESERVED_WORDS_JAVA.indexOf(this.name.toUpperCase()) != -1) {
                 this.env.error(chalk.red('The entity name cannot contain a Java reserved keyword'));
-            } else if (prodDatabaseType == 'mysql' && RESERVED_WORDS_MYSQL.indexOf(this.name.toUpperCase()) != -1) {
-                this.env.error(chalk.red('The entity name cannot contain a MySQL reserved keyword'));
-            } else if (prodDatabaseType == 'postgresql' && RESERVED_WORDS_POSGRES.indexOf(this.name.toUpperCase()) != -1) {
-                this.env.error(chalk.red('The entity name cannot contain a PostgreSQL reserved keyword'));
-            } else if (prodDatabaseType == 'cassandra' && RESERVED_WORDS_CASSANDRA.indexOf(this.name.toUpperCase()) != -1) {
-                this.env.error(chalk.red('The entity name cannot contain a Cassandra reserved keyword'));
-            } else if (prodDatabaseType == 'oracle' && RESERVED_WORDS_ORACLE.indexOf(this.name.toUpperCase()) != -1) {
-                this.env.error(chalk.red('The entity name cannot contain a Oracle reserved keyword'));
-            } else if (prodDatabaseType == 'oracle' && _s.underscored(this.name).length > 26) {
-                this.env.error(chalk.red('The entity name is too long for Oracle, try a shorter name'));
-            } else if (prodDatabaseType == 'mongodb' && RESERVED_WORDS_MONGO.indexOf(this.name.toUpperCase()) != -1) {
-                this.env.error(chalk.red('The entity name cannot contain a MongoDB reserved keyword'));
+            } else if (_.isUndefined(this.tableName)) {
```

Comment:
Not necessarily the table name variable is assigned based on either a passed param or the entity name so the validation happens gor table name in either case.  Take a closer look at the commit